packages/cletus/src/operations/architect.tsx

```diff
@@ -311,3 +315,626 @@ export const type_create = operationOf<
+              });
+            }
+            
+            // Read and process file
+            const parsed = await processFile(fullPath, file.file, {
+              assetPath: await getAssetPath(true),
+              sections: true,
+              transcribeImages: false,
+              describeImages: false,
+              extractImages: false,
+              summarize: false,
+            });
+            
+            // Store sections for smarter splitting
+            const fileData = {
+              file: fullPath, // Use fullPath instead of file.file
+              sections: parsed.sections,
+            };
+            
+            // Calculate total size of sections
+            const sectionsSize = parsed.sections.reduce((sum, s) => sum + s.length, 0);
+            
+            // Add to queue
+            dataQueue.push(fileData);
+            queuedSize += sectionsSize;
+            
+            // Notify processor that data is available
+            if (queueHasData) {
+              const resolver = queueHasData;
+              queueHasData = null;
+              resolver();
+            }
+            
+          } catch (error) {
+            log(`Warning: Failed to process file ${file.file}: ${(error as Error).message}`);
+          }
+        }
+      } catch (error) {
+        parsingError = error as Error;
+      } finally {
+        parsingComplete = true;
+        // Wake up processor if it's waiting
+        if (queueHasData) {
+          const resolver = queueHasData;
+          queueHasData = null;
+          resolver();
+        }
+      }
+    })();
+    
+    // Processor promise - consumes queue and calls AI
+    const processorPromise = (async () => {
+      let filesProcessed = 0;
+      let currentBatch = '';
+      let currentBatchFiles: string[] = [];
+      
+      while (true) {
+        // Wait for data if queue is empty and parsing is ongoing
+        while (dataQueue.length === 0 && !parsingComplete) {
+          await new Promise<void>(resolve => {
+            queueHasData = resolve;
+          });
+        }
+        
+        // Exit if queue is empty and parsing is done
+        if (dataQueue.length === 0 && parsingComplete) {
+          break;
+        }
+        
+        // Process available data
+        while (dataQueue.length > 0) {
+          const fileData = dataQueue.shift()!;
+          const sectionsSize = fileData.sections.reduce((sum, s) => sum + s.length, 0);
+          queuedSize -= sectionsSize;
+          
+          // Notify parser that space is available
+          if (queueHasSpace && queuedSize < maxQueueSize) {
+            const resolver = queueHasSpace;
+            queueHasSpace = null;
+            resolver();
+          }
+          
+          // Process sections smartly - add sections one by one, batching when needed
+          for (const section of fileData.sections) {
+            const sectionContent = `\n\n=== File: ${fileData.file} ===\n${section}`;
```

Comment:
Only add file header if it's the first section of this file being incorporated into the next prompt call